src/crypto/Lyra2RE/bmw.c

```diff
@@ -609,6 +609,7 @@ static const sph_u32 final_s[16] = {
 static void
 bmw32_init(sph_bmw_small_context *sc, const sph_u32 *iv)
 {
+	memset(sc, 0, sizeof(sph_bmw_small_context));
```

Comment:
I think it can have some side effects using `memset` only (even if they are rare case).
How about this?
 
```suggestion
	memset(sc, 0, sizeof(sph_bmw_small_context));
	*sc = { 0 };
```

Reply:
I don't think you can do that.

```
bmw.c: In function ‘bmw32_init’:
bmw.c:612:15: error: expected expression before ‘{’ token
  612 |         *sc = { 0 };
      |               ^
make: *** [<builtin>: bmw.o] Error 1
```

This would work but look less nice:

```c
*sc = (sph_bmw_small_context){ 0 };
```

I feel however the memset is more than appropriate considering the structure of `sph_bmw_small_context` and `sph_bmw_big_context` only contains integers and arrays of chars or integers, and those structures aren't going to change anytime soon.

Reply:
I see. 

> This would work but look less nice:

I agree with your comment. I reject my suggestion.